src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -363,12 +363,13 @@ public boolean queryKeyValuePresence(String key, String value, Set<String> ignor
             ResultSequence rs = session.submitRequest(request);
 
             List<Map<String, Serializable>> projections = new ArrayList<>(limit);
+            DBSStateFlattener flattener new DBSStateFlattener();
             for (String rsItem : rs.asStrings()) {
                 State state = MarkLogicStateDeserializer.deserialize(rsItem);
                 if (manualProjection) {
                     projections.addAll(evaluator.matches(state));
                 } else {
-                    projections.add(DBSStateFlattener.flatten(state));
+                    projections.add(flattener.flatten(state));
```

Comment:
ok the changeset matches the git comment, still I don't understand the intend of the change and how it fixes a random, 
don't hesitate to share your new skills in the git comment or jira ticket

Reply:
The failures were not random, all DBS backends were failing.

Reply:
The main PR is https://github.com/nuxeo/nuxeo/pull/1295